src/gui/MainWindow.cpp

```diff
@@ -208,12 +206,10 @@ MainWindow::MainWindow() :
 
 MainWindow::~MainWindow()
 {
-	for( QList<PluginView *>::iterator it = m_tools.begin();
-						it != m_tools.end(); ++it )
+	for( PluginView *view : m_tools )
 	{
-		Model * m = ( *it )->model();
-		delete *it;
-		delete m;
+		delete view;
+		delete view->model();
```

Comment:
On 01/20/2015 04:38 PM, Lukas W wrote:
>
> In src/gui/MainWindow.cpp
> <https://github.com/LMMS/lmms/pull/1661#discussion-diff-23225530>:
>
> >  	{
> > -		Model * m = ( *it )->model();
> > -		delete *it;
> > -		delete m;
> > +		delete view;
> > +		delete view->model();
>
> This is likely going to segfault, as you're trying to use the member
> function |model()| of a deleted object.
>

May not actually segfault (since the memory doesn't get immediately
overwritten) but it's still bad code, at the very least causes a warning
in valgrind.